tests/test_registrations.py

```diff
@@ -98,52 +131,61 @@ async def save_unclaimed_hospital(self, **kwargs) -> UnclaimedHospital:
 # we need to mock the hospital unit of work.
 # **************************************************** #
 class FakeHospitalUOWAsyncImpl(InterfaceHospitalUOW):
-    hospital_repo: InterfaceHospitalRepo = FakeHospitalRepoImpl()
+    hospital_repo = FakeHospitalRepoImpl()
 
-    def __init__(self):
+    def __init__(self) -> None:
         print("Creating FakeDBSession.")
         self._db_session = FakeDBSession()
         print("Created FakeDBSession.")
 
-    async def commit(self) -> UOWSessionFlag.COMMITTED:
+    async def commit(self) -> Literal[UOWSessionFlag.COMMITTED]:
         """Commit the unit of work."""
         print("Committing unit of work")
         return UOWSessionFlag.COMMITTED
 
-    async def rollback(self) -> UOWSessionFlag.ROLLED_BACK:
+    async def rollback(self) -> Literal[UOWSessionFlag.ROLLED_BACK]:
         """Rollback the unit of work."""
         print("Rolling back unit of work")
         return UOWSessionFlag.ROLLED_BACK
 
-    async def close(self):
+    async def close(self) -> Literal[UOWSessionFlag.CLOSED]:
         """Close the unit of work."""
         print("Closing UOW repo session")
         await asyncio.sleep(2)
+        return UOWSessionFlag.CLOSED
 
     async def __aenter__(self) -> FakeHospitalUOWAsyncImpl:
         """Create a storage session using unit of work."""
         try:
             print("Setting db_session to UOW repo.")
-            self.hospital_repo.session = self._db_session
+            self.hospital_repo.set_session(self._db_session)
             return self
         except (AttributeError, pydantic.ValidationError) as e:
             await self.close()
             raise e
 
-    async def __aexit__(self, exc_type, exc_val, exc_tb):
+    async def __aexit__(
+        self,
+        exc_type: Exception,
+        exc_val: str | MissingRegistrationFieldError,
+        exc_tb: str,
+    ) -> None:
         """Exit context manager."""
         await self.commit()
         if exc_val:
             print("Error during UOW exit.")
+            print(exc_type)
```

Comment:
why we are using print? Can we think about logging instead?

Reply:
Maybe we can check this one? https://docs.pytest.org/en/latest/how-to/logging.html#live-logs

Reply:
Makes somewhat sense.
I have made the changes in the next commit which you can test with the following by appending it to the test file and running `poetry run pytest -m fast tests`:

```python3

@pytest.mark.fast
class TestDummyLogger:
    """Tests the dummy logger."""
    def test_fail(self) -> None:
        try:
            a = 1/0
            assert False
        except ZeroDivisionError as e:
            TEST_LOGGER.error("logging error")
            TEST_LOGGER.exception("LOGGING ERRORRRR", exc_info=e, stack_info=True)
            raise e
```

We don't want to support [live logs from cli](https://docs.pytest.org/en/latest/how-to/logging.html#incompatible-changes-in-pytest-3-4) and keep supporting configuration led and/or programmed test setups.